src/EditorFeatures/CSharpTest/CodeActions/IntroduceVariable/IntroduceVariableTests.cs

```diff
@@ -2345,5 +2345,82 @@ void M()
 }
 ");
         }
+
+        [WorkItem(1130990)]
+        [Fact, Trait(Traits.Feature, Traits.Features.CodeActionsIntroduceVariable)]
+        public void InParentConditionalAccessExpressions()
+        {
+            var code =
+    @"using System;
+class C
+{
+    public T F<T>(T x)
+    {
+        var y = [|F(new C())|]?.F(new C())?.F(new C());
+        return x;
+    }
+}";
+
+            var expected =
+    @"using System;
+class C
+{
+    public T F<T>(T x)
+    {
+        var {|Rename:c|} = F(new C());
+        var y = c?.F(new C())?.F(new C());
+        return x;
+    }
+}";
+
+            Test(code, expected, index: 0, compareTokens: false);
+        }
+
+        [WorkItem(1130990)]
+        [Fact, Trait(Traits.Feature, Traits.Features.CodeActionsIntroduceVariable)]
+        public void MissingAcrossMultipleParentConditionalAccessExpressions()
+        {
+            TestMissing(
+    @"using System;
+class C
+{
+    public T F<T>(T x)
+    {
+        var y = [|F(new C())?.F(new C())|]?.F(new C());
```

Comment:
Extract Method doesn't have specific knowledge about ConditionalAccess expressions today either (invoking on ```[|a.b|].c``` extracts ```a.b``` while invoking on ```[|a?.b|]?.c``` extracts ```a?.b?.c```), but as conditional access expressions gain more usage I suspect that the differences in behavior for all of these features will become frustrating. It would be nice if there was some shared code that could rewrite a given ConditionalAccessExpression to group together some left-most "sub-expression" that these features could all use. If don't update IntroduceVariable to support these cases as part of this fix, let's at least track this additional work in a separate public issue.